src/contracts/pods/EigenPod.sol

```diff
@@ -672,7 +673,14 @@ contract EigenPod is IEigenPod, Initializable, ReentrancyGuardUpgradeable, Eigen
 
         validatorInfo.restakedBalanceGwei = 0;
         validatorInfo.status = VALIDATOR_STATUS.WITHDRAWN;
-        validatorInfo.mostRecentBalanceUpdateTimestamp = withdrawalTimestamp;
+        /**
+        * if multiple full withdrawals are proven for the same validator, we ensure the latest timestamp is recorded
+        * if the withdrawals are proven out of order. A validator may have to make multiple full withdrawals if 
+        * they redeposit into their exited validator.
+        */
+        if (validatorInfo.mostRecentBalanceUpdateTimestamp < withdrawalTimestamp){
+            validatorInfo.mostRecentBalanceUpdateTimestamp = withdrawalTimestamp;   
+        }
```

Comment:
so this change is specifically in reference to https://github.com/Layr-Labs/eigenlayer-contracts/issues/240 -- right? 
I commented some on the issue in Slack -- seems like we need to agree on the correct behavior before merging this change.

Reply:
Resolved!